docs/versioned_docs/version-zenith/quickstart/428201-custom-modules.mdx

```diff
@@ -0,0 +1,145 @@
+
+To see this in action, create a custom Dagger module to build a Go binary, add it to a container image and publish the container image to a registry.
+
+1. Begin by creating a new directory for your Dagger module and running `dagger init` to bootstrap a new module:
+
+    <Tabs groupId="language">
+    <TabItem value="Go">
+
+    ```shell
+    mkdir my-module
+    dagger init --name=my-module --sdk=go
+    ```
+
+    This will generate a `dagger.json` module file, an initial `main.go` source file, as well as a generated `dagger.gen.go` and `querybuilder` directory for the generated module code.
+
+    </TabItem>
+    <TabItem value="Python">
+
+    ```shell
+    mkdir my-module
+    dagger init --name=my-module --sdk=python
+    ```
+
+    This will generate a `dagger.json` module file, initial `src/main.py` and `pyproject.toml` files, as well as a generated `sdk` folder for local development.
+
+    </TabItem>
+    <TabItem value="TypeScript">
+
+    ```shell
+    mkdir my-module
+    dagger init --name=my-module --sdk=typescript
+    ```
+
+    This will generate a `dagger.json` module file, initial `src/index.ts`, `package.json` and `tsconfig.json` files, as well as a generated `sdk` folder for local development.
+
+    </TabItem>
+    </Tabs>
+
+1. Dagger lets you add dependencies to your modules, so that you can build on modules created by others. Add the `golang` and `wolfi` modules as dependencies in your new module using `dagger install`:
+
+    ```shell
+    dagger install github.com/kpenfound/dagger-modules/golang
+    dagger install github.com/shykes/daggerverse/wolfi
+    ```
+
+1. Update the generated `main.go` file with the following code:
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+    ```go file=./snippets/create/go/main.go
+    ```
+
+    This new `BuildAndPublish()` function accepts three arguments - the source directory, a list of build arguments and the path to the compiled binary file - and does the following:
+
+    - It invokes the `Golang` module via the `dag` Dagger client.
+    - It calls the module's `WithProject()` and `Build()` functions to build the remote directory. As you've seen already, the `Build()` function returns a `Directory` representing the state of the filesystem following the build.
+    - It calls the `Directory.File()` method to return a reference to the compiled binary file.
+    - It invokes the `Wolfi` module via the `dag` client.
+    - It calls the module's `Base()` and `Container()` functions to return a new Wolfi container image as a `Container`.
+    - It calls the `Container.WithFile()` method to add the compiled binary file to the container image's filesystem. This method returns a new `Container` with the result.
+    - It calls the `Container.Publish()` method to publish the container image to the registry and print the SHA identifier of the published image.
+
+</TabItem>
+<TabItem value="Python">
+
+    ```python file=./snippets/create/python/main.py
```

Comment:
I see this error when calling the function, only in Python

```
Function execution error: unhandled errors in a TaskGroup (1 sub-exception) 
```

Can you suggest the fix @helderco ?

Reply:
That's not the actual error. Need to see the sub-exception so I need to run it and see.

Reply:
The actual error was:
```
error: parse selections: parse field "build": init arg "args" value as dagql.DynamicArrayInput ([String!]! using dagql.DynamicArrayInput: decode "./cmd/dagger": invalid character '.' looking for beginning of value
```

So:
```diff
-build_args: str
+build_args: list[str]
```

Reply:
I pushed the fix.